test/cri-containerd/container_update_test.go

```diff
@@ -0,0 +1,364 @@
+					},
+					Image: &runtime.ImageSpec{
+						Image: test.containerImage,
+					},
+					Command: test.cmd,
+				},
+				PodSandboxId:  podID,
+				SandboxConfig: podRequest.Config,
+			}
+
+			containerID := createContainer(t, client, ctx, containerRequest)
+			defer removeContainer(t, client, ctx, containerID)
+
+			startContainer(t, client, ctx, containerID)
+			defer stopContainer(t, client, ctx, containerID)
+
+			// make request to increase cpu shares
+			updateReq := &runtime.UpdateContainerResourcesRequest{
+				ContainerId: containerID,
+			}
+
+			var expected uint32 = 5000
+			if test.runtimeHandler == lcowRuntimeHandler {
+				updateReq.Linux = &runtime.LinuxContainerResources{
+					CpuShares: int64(expected),
+				}
+			} else {
+				updateReq.Windows = &runtime.WindowsContainerResources{
+					CpuShares: int64(expected),
+				}
+			}
+
+			if _, err := client.UpdateContainerResources(ctx, updateReq); err != nil {
+				t.Fatalf("updating container resources for %s with %v", containerID, err)
+			}
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				checkLCOWResourceLimit(t, ctx, client, containerID, "/sys/fs/cgroup/cpu/cpu.shares", uint64(expected))
+			} else if test.runtimeHandler == wcowHypervisorRuntimeHandler {
+
+				targetShimName := "k8s.io-" + podID
+				jobExpectedValue := jobobject.CalculateJobCPUWeight(expected)
+				checkWCOWResourceLimit(t, ctx, test.runtimeHandler, targetShimName, containerID, "cpu-weight", uint64(jobExpectedValue))
+			}
+		})
+	}
+}
+
+func Test_Container_UpdateResources_CPUShare_NotRunning(t *testing.T) {
+	type config struct {
+		name             string
+		requiredFeatures []string
+		runtimeHandler   string
+		sandboxImage     string
+		containerImage   string
+		cmd              []string
+	}
+	tests := []config{
+		{
+			name:             "WCOW_Process",
+			requiredFeatures: []string{featureWCOWProcess},
+			runtimeHandler:   wcowProcessRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "WCOW_Hypervisor",
+			requiredFeatures: []string{featureWCOWHypervisor},
+			runtimeHandler:   wcowHypervisorRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "LCOW",
+			requiredFeatures: []string{featureLCOW},
+			runtimeHandler:   lcowRuntimeHandler,
+			sandboxImage:     imageLcowK8sPause,
+			containerImage:   imageLcowAlpine,
+			cmd:              []string{"top"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, []string{test.sandboxImage})
+			} else if test.runtimeHandler == wcowHypervisorRuntimeHandler {
```

Comment:
Only if it's hypervisor? We seem to be running these for process and hv so why only pull the Windows image for the hv runtime?